src/work-allocation/components/task-assignment/task-assignment.component.ts

```diff
@@ -103,6 +116,34 @@ export class TaskAssignmentComponent implements OnInit {
     this.handleLocationChanged();
   }
 
+  /**
+   * Sets the logged in user id
+   */
+  public setUserId(): void {
+    const userInfoStr = this.sessionStorageService.getItem('userDetails');
+    if (userInfoStr) {
+      const userInfo: UserInfo = JSON.parse(userInfoStr);
+      this.userId = userInfo.id;
+    }
+    return undefined;
+  }
+
+  /**
+   * Using set user id, gets caseworker details for the logged in caseworker
+   */
+  public setAssignedCaseworker(): void {
+    this.caseworkerService.getAll().subscribe(caseworkers => {
+      const assignedCaseworker = caseworkers.find(cw => this.isAssignedCaseworker(cw.idamId));
+      this.assignedCaseworker = assignedCaseworker;
+    }, error => {
+      handleFatalErrors(error.status, this.router, WILDCARD_SERVICE_DOWN);
+    });
+  }
+
+  public isAssignedCaseworker(idamId: string): boolean {
+    return idamId === this.userId ? true : false
```

Comment:
No need for this to be a ternary operator: `return idam === this.userId;` is sufficient.

Reply:
Yes, I noticed that and changed it already ten mins ago when I was looking through code. It was probably both a remnant of checking that the userId was present and a lapse in concentration